include/fs/fs.h

```diff
@@ -160,6 +160,30 @@ struct fs_file_system_t {
 					struct fs_statvfs *stat);
 };
 
+#ifndef FO_MODE_BITS
+#define FO_MODE_BITS	0x03
+#endif
+
+#ifndef FO_READ
```

Comment:
All these `#ifndef`s are weird: why would somebody define `FO_READ` before this header, and if they did wouldn't it be a problem?  That's a motivation for using an enum for the constants.

IMO these bits should be defined as `FS_MODE_READ` etc, preserving the existing `fs_` prefix for the public API in this header.  It could be `FS_OPEN_READ` if there's no possibility of using these for other roles (such as permissions).

Reply:
Ifdefs removed. I am not sure about `FS_` prefix. I have selected `FO_` for the file open, and this is similar to posix where open flags start with `O_`

Reply:
API 2020-07-07 Per discussion prefer `FS_O_foo` to be namespaced in filesystem API but match POSIX `O_foo` flag names.

Reply:
Done.